build/mvn

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
```

Comment:
Taken from https://github.com/apache/spark/blob/master/build/mvn

Reply:
Why do we need this? We'll need to ensure that it stays in sync with our repo? Or maybe we can get rid of it once we merge this upstream?

Reply:
We get rid of this when we merge into upstream, but even aside from that, this has hardly been changing in upstream and it's theoretically completely isolated from what we will use to build Spark (since the shell forks the second Maven process it's completely independent)

Reply:
We also have to add it here now because unlike before, now `mvn` is the entrypoint into the whole system, as it's only the substep of the build reactor that clones the Spark repository down. Aside from that, we also allow for Spark TGZs to be provided to remove the usage of the Spark source code entirely. So no matter what we're going to need our own Maven here.

Reply:
Instead of copying the content here, could you just have a much shorter script that just downloads from https://raw.githubusercontent.com/apache/spark/master/build/mvn and execs it? Something like build/mvn-getter whose content is:
```
#!/bin/sh
curl -s https://raw.githubusercontent.com/apache/spark/master/build/mvn | bash

Reply:
Hm, I think that works. Will incorporate that.

Reply:
Done